### PR TITLE
Make handlerSettings optional

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/IISIntegration/issues/988

### CODE REVIEW

File: test/CommonLibTests/ConfigUtilityTests.cpp [hunk 1]
```diff
@@ -89,4 +89,24 @@ namespace ConfigUtilityTests
...
     }
+
+    TEST(ConfigUtilityTestSingle, IgnoresFailedGetElement)
+    {
+        IAppHostElement* retElement = NULL;
+        STRU handlerVersion;
+
+        auto element = std::make_unique<NiceMock<MockElement>>();
+        auto innerElement = std::make_unique<NiceMock<MockElement>>();
+        auto collection = std::make_unique<NiceMock<MockCollection>>();
```
Comment (jkotalik) [inline test/CommonLibTests/ConfigUtilityTests.cpp]: remove some unused ones.

File: test/CommonLibTests/ConfigUtilityTests.cpp [hunk 2]
```diff
@@ -89,4 +89,18 @@ namespace ConfigUtilityTests
...
+
+    TEST(ConfigUtilityTestSingle, IgnoresFailedGetElement)
+    {
+        STRU handlerVersion;
+
+        auto element = std::make_unique<NiceMock<MockElement>>();
+        ON_CALL(*element, GetElementByName(_, _))
+            .WillByDefault(DoAll(testing::SetArgPointee<1>(nullptr), testing::Return(HRESULT_FROM_WIN32( ERROR_INVALID_INDEX ))));
+
+        HRESULT hr = ConfigUtility::FindHandlerVersion(element.get(), &handlerVersion);
```
Comment (jkotalik) [inline test/CommonLibTests/ConfigUtilityTests.cpp]: Anyway to check the log in this test?

Comment (pakrym) [reply test/CommonLibTests/ConfigUtilityTests.cpp]: There might be by collecting stdout but I don't think it's required here.

**Comment by mikeharder** [comment]
@jkotalik : Created issue to add tests for invalid schema (https://github.com/aspnet/IISIntegration/issues/992).

**Comment by pakrym** [comment]
Added a fix for event log event tracing and file sharing on AppServices.

**Pull request merged**